packages/pwa-kit-react-sdk/src/ssr/universal/contexts/index.js

```diff
@@ -7,7 +7,43 @@
 
 import React from 'react'
 import DeviceContext from '../device-context'
+import {v4 as uuidv4} from 'uuid'
 
 const ExpressContext = React.createContext()
 
-export {DeviceContext, ExpressContext}
+export {CorrelationIdContext}
+
+const CorrelationIdContext = React.createContext()
+
+const CorrelationIdProvider = ({children, req}) => {
+    let correlationId
+
+    if (req) {
```

Comment:
I'd rethink using the request as a param for this provider. The reason I say this is because it creates an unwarranted dependency on having a request object. If for example we weren't getting the correlation id from the request we wouldn't need it. I'd suggest just passing in a correlationId as a prop. And leave the logic for determining it's value for the server and main files.